tibanna_ffcommon/portal_utils.py

```diff
@@ -1220,10 +1242,80 @@ def update_input_extras(self):
                     self.update_patch_items(ip['uuid'], {'higlass_uid': higlass_uid})
             self.update_patch_items(ip['uuid'], {'extra_files': ip['extra_files']})
 
+
+    def update_generic_qc(self):
+
+        input_file_args = self.workflow_arguments([INPUT_FILE])
+        generic_qc_args = self.workflow_arguments([GENERIC_QC_FILE])
+
+        if len(generic_qc_args) == 0:
+            return
+
+        # Basic sanity checks of the workflow arguments
+        check_qc_workflow_args(input_file_args, generic_qc_args)
+
+        ff_key = self.tibanna_settings.ff_keys
+        ff_env = self.tibanna_settings.env
+
+        for input_file_arg in input_file_args:
+            # Get the associated QC args
+            input_wf_arg_name = input_file_arg['workflow_argument_name']
+            input_file_accession = self.accessions(input_wf_arg_name)[0]
+            qc_args = filter_workflow_args_by_property(generic_qc_args, "argument_to_be_attached_to", input_wf_arg_name)
+            if len(qc_args) == 0: # Input file has no corresponding Generic QC files
+                continue
+
+            qc_args_json = filter_workflow_args_by_property(qc_args, "qc_json", True)
+            # After running check_qc_workflow_args, we know that this contains exactly one element
+            qc_arg_json = qc_args_json[0]
+            qc_arg_json_name = qc_arg_json['workflow_argument_name']
+            try:
+                qc_json = self.read_json_from_s3(qc_arg_json_name)
+            except Exception as e:
+                logger.error(str(e))
+                raise GenericQcException(f"Could not get {qc_arg_json_name} from S3. Error: {str(e)}")
+
+            # Get the link to the zipped report. This will be added to the QualityMetricGeneric item
+            # After running check_qc_workflow_args, we know that this contains zero or one elements
+            qc_args_zipped = filter_workflow_args_by_property(qc_args, "qc_zipped", True)
+            qc_arg_zipped = qc_args_zipped[0] if len(qc_args_zipped) == 1 else None
+            qc_arg_zipped_s3_url = f"https://{self.outbucket}.s3.amazonaws.com/{self.file_key(qc_arg_zipped['workflow_argument_name'])}" if qc_arg_zipped else None
+
+            # The folling will create a new QualityMetricGeneric item in the portal
+            qmg_uuid = str(uuid4())
+            try:
+                post_dict = {
+                    'uuid': qmg_uuid,
+                    'institution': self.ff_meta.institution,
+                    'project': self.ff_meta.project,
+                }
```

Comment:
This is CGAP specific (project/institution) no? So does this really belong in tibanna_ffcommon? How would this work in SMaHT?

Reply:
This has been generalized.